commands/debug/debug_test.go

```diff
@@ -0,0 +1,95 @@
+// This file is part of arduino-cli.
+//
+// Copyright 2020 ARDUINO SA (http://www.arduino.cc/)
+//
+// This software is released under the GNU General Public License version 3,
+// which covers the main part of arduino-cli.
+// The terms of this license can be found at:
+// https://www.gnu.org/licenses/gpl-3.0.en.html
+//
+// You can be released from the requirements of the above licenses by purchasing
+// a commercial license. Buying such a license is mandatory if you want to
+// modify or otherwise use the software for commercial activities involving the
+// Arduino software without disclosing the source code of your own applications.
+// To purchase a commercial license, send an email to license@arduino.cc.
+package debug
+
+import (
+	"fmt"
+	"runtime"
+	"strings"
+	"testing"
+
+	"github.com/arduino/arduino-cli/arduino/cores/packagemanager"
+	dbg "github.com/arduino/arduino-cli/rpc/debug"
+	"github.com/arduino/go-paths-helper"
+	"github.com/stretchr/testify/assert"
+)
+
+var customHardware = paths.New("testdata", "custom_hardware")
+var dataDir = paths.New("testdata", "data_dir", "packages")
+var sketch = "hello"
+var sketchPath = paths.New("testdata", sketch)
+
+func TestGetCommandLine(t *testing.T) {
+	pm := packagemanager.NewPackageManager(nil, nil, nil, nil)
+	pm.LoadHardwareFromDirectory(customHardware)
+	pm.LoadHardwareFromDirectory(dataDir)
+
+	// Arduino Zero has an integrated debugger port, anc it could be debugged directly using USB
+	req := &dbg.DebugConfigReq{
+		Instance:   &dbg.Instance{Id: 1},
+		Fqbn:       "arduino-test:samd:arduino_zero_edbg",
+		SketchPath: sketchPath.String(),
+		Port:       "none",
+	}
+
+	goldCommand := fmt.Sprintf("%s/arduino-test/tools/arm-none-eabi-gcc/7-2017q4/bin//arm-none-eabi-gdb", dataDir) +
+		" -ex target extended-remote |" +
+		fmt.Sprintf(" %s/arduino-test/tools/openocd/0.10.0-arduino7/bin/openocd", dataDir) +
+		fmt.Sprintf(" -s \"%s/arduino-test/tools/openocd/0.10.0-arduino7/share/openocd/scripts/\"", dataDir) +
+		fmt.Sprintf(" --file \"%s/arduino-test/samd/variants/arduino_zero/openocd_scripts/arduino_zero.cfg\"", customHardware) +
+		fmt.Sprintf(" -c \"gdb_port pipe\" -c \"telnet_port 0\" -c init -c halt %s/hello.arduino-test.samd.arduino_zero_edbg.elf", sketchPath)
+
+	if runtime.GOOS == "windows" {
```

Comment:
You could build all the paths with `/` and then call https://golang.org/pkg/path/filepath/#FromSlash which will provide the right separators for each platform

Reply:
Windows is able to handle paths with forward slash `/`, even mixed with `\`, there is no need to use `FromSlash` or `ToSlash` (we used it in the past, before discovering this nice fact :-))

Reply:
I have used `FromSlash` to sanitize the golden path and the generated path, this way the test is simpler, Anyway a "windows" switch is necessary, because of `.exe` extension for tools binaries

Reply:
Turned out that to successfully use this pattern we had to turn everything to **forward** slashes `/`...
https://github.com/arduino/arduino-cli/pull/597/files#diff-ec656dd0bf60c33c4e4144e7d5e9508aR57

more on that is coming...